### DISCUSSION TITLE
pyscopg-binary is not recommended for production workloads

### DISCUSSION THREAD

**Discussion by billcrook** [fastapi/full-stack-fastapi-template#2135] [Questions] [answered]
See https://www.psycopg.org/docs/install.html#psycopg-vs-psycopg-binary

I was able to build in 3.8 slim by adding:
* build-time dependencies:
  *  `libpq-dev='11.12-0+deb10u1'`
  * `build-essential='12.6'`
* run-time dependencies:
  * `libpq5='11.12-0+deb10u1'`

The build-time packages can be removed from the docker image after psycopg2 python package is installed.

**Comment by tiangolo** [comment, marked as answer]
Thanks, it seems the main reason why it's not recommended is if the system underneath upgrades the versions of the libraries it bundles. Nevertheless if you upgrade the version of pyscopg-binary, it would come with the latest version that it was built with, so I would think that's comparable.

Also, I normally wouldn't want to suggest or encourage users to build dependencies from source, that's can cause so many problems. But I think people that want to set up this way would be perfectly able to do so as well. :nerd_face:
